shared/client/util.go

```diff
@@ -0,0 +1,47 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/chanzuckerberg/happy-shared/model"
+	"github.com/pkg/errors"
+)
+
+var ErrRecordNotFound = errors.New("record not found")
+
+func ParseResponse[T interface{}](resp *http.Response, result *T) error {
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "failed to read response body")
+	}
+
+	err = json.Unmarshal(body, &result)
+	if err != nil {
+		return errors.Wrap(err, "failed to unmarshal response body")
+	}
+
+	return nil
+}
+
+func InspectForErrors(resp *http.Response) error {
+	if resp.StatusCode == http.StatusNotFound {
+		return ErrRecordNotFound
+	} else if resp.StatusCode == http.StatusBadRequest {
+		validationErrors := []model.ValidationError{}
```

Comment:
if you make the `model.ValidationError` implement the error interface, you could use the multierror package to format these errors in a little bit cleaner:

type error interface {
    Error() string
}

Reply:
Nice, good call 👍